girder/girder_large_image/rest/large_image_resource.py

```diff
@@ -445,6 +447,70 @@ def _deleteCachedImages(self, spec, associatedImages=False, imageKey=None):
+        .param('createJobs', 'Whether job(s) should be created for each large image.',
+               required=False, default=False, dataType='boolean')
+        .param('localJobs', 'Whether the job(s) created should be local.', required=False,
+               default=False, dataType='boolean')
+        .param('recurse', 'Whether child folders should be recursed.', required=False,
+               default=False, dataType='boolean')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Write access was denied for the folder.', 403),
+    )
+    def createLargeImages(self, folder, params):
+        user = self.getCurrentUser()
+        createJobs = params.get('createJobs')
+        if createJobs:
+            createJobs = 'always' or True
+        return self.createImagesRecurseOption(folder=folder, createJobs=createJobs, user=user,
+                                              recurse=params.get('recurse'),
+                                              localJobs=params.get('localJobs'))
+
+    def createImagesRecurseOption(self, folder, createJobs, user, recurse, localJobs):
+        result = {'childFoldersRecursed': 0,
+                  'largeImagesCreated': 0,
+                  'largeImagesRemovedAndRecreated': 0,
+                  'totalItems': 0}
+        if recurse:
+            for childFolder in Folder().childFolders(parent=folder, parentType='folder'):
+                result['childFoldersRecursed'] += 1
+                childResult = self.createImagesRecurseOption(folder=childFolder,
+                                                             createJobs=createJobs, user=user,
+                                                             recurse=recurse, localJobs=localJobs)
+                for key in childResult:
+                    result[key] += childResult[key]
+        for item in Folder().childItems(folder=folder):
+            result['totalItems'] += 1
+            if item.get('largeImage'):
+                if item['largeImage'].get('expected'):
+                    pass
+                else:
+                    try:
+                        Item().getMetadata(item)
+                        continue
+                    except Exception:
+                        previousFileId = item['largeImage'].get(
+                            'originalId', item['largeImage']['fileId'])
+                        ImageItem().delete(item)
+                        ImageItem().createImageItem(item, File().load(user=user, id=previousFileId),
+                                                    createJob=createJobs, localJob=localJobs)
+                        result['largeImagesRemovedAndRecreated'] += 1
+            else:
+                largeImageFileId = None
+                files = list(Item().childFiles(item=item))
+                if len(files) == 1:
+                    largeImageFileId = str(files[0]['_id'])
+                    ImageItem().createImageItem(item, File().load(user=user, id=largeImageFileId),
```

Comment:
These lines can be simplied -- there is no need to load the file; we already have it.  That ism you can get rid of `largeImageFileId = str(files[0]['_id'])` and just do `ImageItem().createImageItem(item, files[0], createJob=createJobs, localJob=localJobs)`

Reply:
That's much cleaner. Change made [here](https://github.com/girder/large_image/pull/1572/commits/71107ba9aa59ce3d162431c3bee4b1144444c44e).